src/typescript/frontend/src/components/pages/wallet/useSwapEventsQuery.ts

```diff
@@ -25,7 +37,7 @@ export const useSwapEventsQuery = (args: z.input<typeof GetTradesSchema>, disabl
     getNextPageParam: (lastPageResponse, allPageResponses) =>
       lastPageResponse?.length === LIMIT ? allPageResponses.length + 1 : undefined,
     // Disable the query when neither sender nor marketID is provided
-    enabled: !disabled && (!!args.sender || !!args.marketID),
+    enabled: options?.enabled !== false && (!!args.sender || !!args.marketID),
```

Comment:
```suggestion
    enabled: options?.enabled === true && (!!args.sender || !!args.marketID),
```

Should be this, maybe? Or are you requiring an explicit `enabled === false` to disable it?

If so I'd change it to be named something more explicit- in general though it's not ideal to treat `undefined` similar to a truthy value, would rather fix the names and/or behavior to be more intuitive